### PR TITLE
Fix EMCC_FORCE_STDLIBS=1

### PR DESCRIPTION
**Pull request by kripken**
"1" means "all the libraries", but we didn't use that correctly in all places, and it didn't always link in everything you wanted. This PR expands out "1" into the list of all libraries, so later uses of the list work as expected.

Also add a little text to some error messages, and together these enable `test_dylink_syslibs` to work in wasm.

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -616,9 +607,21 @@ class Dummy(object):
 
   libs_to_link = []
   already_included = set()
-
   system_libs_map = {l.shortname: l for l in system_libs}
 
+  # Setting this in the environment will avoid checking dependencies and make building big projects a little faster
```
Comment (dschuff) [inline tools/system_libs.py]: This means that it forces it to build all stdlibs on demand, rather than just those needed by this compile?

Comment (kripken) [reply tools/system_libs.py]: Yes, exactly.

**Comment by dschuff** [comment]
So is this logic redundant with the logic in embuilder.py? Can we simplify things?

File: tools/system_libs.py [hunk 2]
```diff
@@ -616,9 +607,21 @@ class Dummy(object):
...
+  # 1 means include everything; otherwise it can be the name of a lib (libc++, etc.)
+  # You can provide 1 to include everything, or a comma-separated list with the ones you want
+  force = os.environ.get('EMCC_FORCE_STDLIBS')
+  if force == '1':
+    force_all = True
```
Comment (sbc100) [inline tools/system_libs.py]: I don't think you need the `force_all` variable anymore since adding the lib to `force` has the same effect.

**Comment by kripken** [comment]
Yeah, agreed we should switch to more standard flags. Looking at the history here, what appears to have happened is that the original meaning of "1" was an escape hatch for when you can't figure out which system libs you need, and then with that flag it included them all. But the comment was not clear enough, and the test was not strict enough, sorry about that, so this was intepreted differently. Then a user had a testcase which actually neeed all system libs, so that escape hatch was needed, which led to me doing this PR.

For now this is ok I think, but yeah, as I said I agree we should remove this and have more standard linking options, once things are more stable.

**Pull request merged**